### ISSUE TITLE
Input static creative xfi pci

### ISSUE THREAD

**Issue by mtmfc** [mdias/rs_asio#231]
Hi
Remember pci? I still use an old creative x-fi XtremeMusic from back in the day.  
Output is clear as can be but the input has a strange static/crackling sound. At the menu for example, any input sound is accompanied by this noise. The game works fine apart from this with a buffer size of 64. I've combed through all the info I can find and have tried every combination of settings but nothing seems to help.
Any ideas? 
Thanks so much for developing this, it is brilliant.

**Comment by mdias** [comment]
Hello,

Can you post the contents of the `RS_ASIO-log.txt` file after a short run where you experience this issue?
Also, have you tried:
- Increasing the ASIO buffer size?
- Check with another ASIO-enabled application to see if it works well there?

**Comment by mtmfc** [comment]
I have attached the log.  
ASIO works fine with other apps and I experience no problems at all with similar settings.  
I've tried increasing the buffer size in RS_ASIO.ini by changing the CustomBufferSize setting (BufferSizeMode=custom) and it has not made any difference.  The only change I've observed is all the audio is distorted below 64.  I have also tried increasing the LatencyBuffer in Rocksmith.ini from 1 to 4 and this doesn't make a difference either.  
Thanks let me know if you need any other info.
[RS_ASIO-log.txt](https://github.com/mdias/rs_asio/files/7563046/RS_ASIO-log.txt)

**Comment by mdias** [comment]
Everything looks good on the log, there are no buffer underruns or anything like that.

Your interface appears to have a lot of loopback input channels though. Are you using the correct input channel?
Can you show also your `RS_ASIO.ini`?

**Comment by mtmfc** [comment]
The card has 3 modes and all these inputs only appear in audio creation mode.  
Here is the RS_ASIO.ini.  
[Config]
EnableWasapiOutputs=0
EnableWasapiInputs=0
EnableAsio=1

[Asio]
; available buffer size modes:
;    driver - respect buffer size setting set in the driver
;    host   - use a buffer size as close as possible as that requested by the host application
;    custom - use the buffer size specified in CustomBufferSize field
BufferSizeMode=custom
CustomBufferSize=64

[Asio.Output]
Driver=Creative ASIO
BaseChannel=0
AltBaseChannel=4
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.0]
Driver=Creative ASIO
Channel=14
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

**Comment by mtmfc** [comment]
I've been playing around with the settings a little - the lowest buffer size without distortion is 56.  Above 112 I get distortion too which I wasn't expecting.  Not sure if that is relevant at all.

**Comment by mdias** [comment]
Your config looks fine.
I went through your log again, and everything is looking pretty normal over there too.

This could be another similar case to the Focusrite where the driver misbehaves on specific buffer sizes when the application is timing sensitive. In the Focusrite case other software works fine too, and the reason for that is probably because other software just double-buffers, while RS_ASIO needs proper real-time feed of the buffers.

To try if this is the case, I suggest you try the following buffer sizes: 64, 96, 128, 160, 192, 224, 256.

Also just for testing set `EnableSoftwareEndpointVolumeControl` and `EnableSoftwareMasterVolumeControl` to zero on both output and input. That should decrease CPU time spent on the buffer swap a tiny bit; could be that it helps too.

**Comment by mtmfc** [comment]
I've tried everything you said there and nothing affected the issue - it doesn't get better or worse, it just seems to stay the same.  Based on this line in the log: ASIOBufferSize - min: 48 max: 32768 preferred: 48 granularity: 8, I tried settings in increments of 8 all the way from 64 up to 128 as well.  
I was able to get buffer sizes above 120 working without output distortion by increasing the latency setting in rocksmith.ini (at 1 usually).  
Any other ideas? Thanks.
